docs: fix blog link to starport.com/blog

barriebyron opened the pull request:
thanks @soniasingla for noticing!

barriebyron commented:
Let's have more descriptive link text, wdyt?

soniasingla reviewed (commented):
@barriebyron under the section `Stay in touch`, it would be nice to add Discord as well. WDYT?

barriebyron commented:
> @barriebyron under the section `Stay in touch`, it would be nice to add Discord as well. WDYT?

We had Discord listed above already, but I added again since I like the suggestion to have a complete list of resources... I added docs, too. I waffled a bit because our video library is so tiny, but I also added YouTube. We'll have the second community call recording to add soon... and more as we grow.
